realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/utils/UserFactory.java

```diff
@@ -161,40 +161,12 @@ public static void logoutAllUsers() {
         handler.post(new Runnable() {
             @Override
             public void run() {
-                final AtomicInteger usersLoggedOut = new AtomicInteger(0);
-                final int activeUsers = SyncUser.all().size();
-                final AuthenticationListener listener = new AuthenticationListener() {
-                    @Override
-                    public void loggedIn(SyncUser user) {
-                        SyncManager.removeAuthenticationListener(this);
-                        fail("User logged in while exiting test: " + user);
-                    }
-
-                    @Override
-                    public void loggedOut(SyncUser user) {
-                        if (usersLoggedOut.incrementAndGet() == activeUsers) {
-                            SyncManager.removeAuthenticationListener(this);
-                            allUsersLoggedOut.countDown();
-                        }
-                    }
-                };
-                SyncManager.addAuthenticationListener(listener);
-
                 Map<String, SyncUser> users = SyncUser.all();
-                if (users.isEmpty()) {
-                    SyncManager.removeAuthenticationListener(listener);
-                    allUsersLoggedOut.countDown();
-                } else {
-                    for (SyncUser user : users.values()) {
-                        user.logout();
-                        if (!user.getAuthenticationUrl().toString().contains("127.0.0.1")) {
-                            // For dummy users, calling `logout()` will never result in the
-                            // authentication listener to trigger since the URL doesn't exist.
-                            // For these cases, we manually trigger the listener.
-                            listener.loggedOut(user);
-                        }
-                    }
+                for (SyncUser user : users.values()) {
+                    user.logout();
                 }
+                SystemClock.sleep(2000); // Remove when https://github.com/realm/ros/issues/304 is fixed
```

Comment:
https://github.com/realm/ros/issues/304 has been marked as fixed

Reply:
Yes, but not released yet.